pkg/networkservice/xconnect/l2bridgedomain/server.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2022 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l2bridgedomain
+
+import (
+	"context"
+
+	"git.fd.io/govpp.git/api"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/payload"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/networkservicemesh/sdk/pkg/tools/postpone"
+)
+
+type l2BridgeDomainServer struct {
+	vppConn api.Connection
+	b       l2BridgeDomain
+}
+
+// NewServer returns a Client chain element that will add client and server vpp interface (if present) to a dridge domain
+func NewServer(vppConn api.Connection) networkservice.NetworkServiceServer {
+	return &l2BridgeDomainServer{
+		vppConn: vppConn,
+	}
+}
+
+func (v *l2BridgeDomainServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	if request.GetConnection().GetPayload() != payload.Ethernet {
+		return next.Server(ctx).Request(ctx, request)
+	}
```

Comment:
How do we tell this case from a non-vlan Ethernet Payload case?

Reply:
At the beginning of  function [addBridgeDomain](https://github.com/Nordix/nsm-sdk-vpp/blob/00dee63c4afe800c110562d62705a41c2a94b82e/pkg/networkservice/xconnect/l2bridgedomain/common.go#L48) there is a check if there is a vlan_id stored for the connection context. The function returns immediately if this is missing. The VLAN ID is stored when the sub-interface is created.

Reply:
Might be good to make this clearer by putting such a check up at the top of request so its clear to the reader when we do or don't carry out this chain.